From 7df5f59f13d55b6d56b0abebfb626263fb43a3ec Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Fri, 10 Oct 2014 12:33:29 +0100
Subject: [PATCH] Add additional indexes to the music tables.

One index on song_id in music_albumart.
One index on artist_id in music_albums.

Speeds up browsing music and loading of the UPnP tree.
---
 mythplugins/mythmusic/mythmusic/dbcheck.cpp |   14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/mythplugins/mythmusic/mythmusic/dbcheck.cpp b/mythplugins/mythmusic/mythmusic/dbcheck.cpp
index 5db3673..d1ca96e 100644
--- a/mythplugins/mythmusic/mythmusic/dbcheck.cpp
+++ b/mythplugins/mythmusic/mythmusic/dbcheck.cpp
@@ -12,7 +12,7 @@ using namespace std;
 
 #include "dbcheck.h"
 
-const QString currentDatabaseVersion = "1022";
+const QString currentDatabaseVersion = "1023";
 
 static bool doUpgradeMusicDatabaseSchema(QString &dbver);
 
@@ -902,5 +902,17 @@ QString("UPDATE music_albumart SET hostname = '%1';").arg(gCoreContext->GetMaste
             return false;
     }
 
+    if (dbver == "1022")
+    {
+        const QString updates[] = {
+"CREATE INDEX `song_id` ON music_albumart (song_id);",
+"CREATE INDEX `artist_id` ON music_albums (artist_id);",
+""
+};
+
+        if (!performActualUpdate(updates, "1023", dbver))
+            return false;
+    }
+
     return true;
 }
-- 
1.7.10.2

